### PR TITLE
from_protobuf: Add reusable protobuf test data helpers [databricks] [reduced-it]

### PR DESCRIPTION
**Pull request by thirtiseven**
Part of #14069.

### Description

This PR adds a reusable protobuf encoding helper for integration tests. It reuses the existing `DataGen` framework (`StructGen` and `gen_df`) to generate logical values, then uses Spark's CPU `to_protobuf` to encode them. The helper also accepts explicit rows and handles descriptor-path and binary-descriptor API variants.

Encoding is a separate preparation step because it requires a Spark SQL operation rather than a per-value Python transformation. Collecting the encoded rows lets subsequent CPU/GPU decoding tests consume identical payloads without including encoding in the comparison.

The helper generates valid payloads from a supplied, matching generator and descriptor. It does not generate descriptors automatically; malformed inputs and other wire-level edge cases still require dedicated fixtures or byte encoders. It introduces no additional Python protobuf dependency and enables no GPU protobuf functionality.

Tests cover API argument forwarding, binary-column name collisions, and CPU encoding/decoding with both explicit rows and generated data. The Spark tests check row counts, preserved logical values, output schema, and that each decoded payload matches its corresponding logical row.

Validation:
- `./integration_tests/run_pyspark_from_build.sh -s -k 'protobuf_data_gen'`
- 5 tests passed on Apache Spark 3.5.2 using `LOCAL_JAR_PATH` with the previously built plugin and integration-test artifacts from the compatibility worktree on the same main baseline.
- `git diff --check` passed.

AI assistance: The change and PR description were prepared with Codex assistance and reviewed by the author before submission.

### Checklists

Documentation
- [ ] Updated for new or modified user-facing features or behaviors
- [x] No user-facing change

Testing
- [x] Added or modified tests to cover new code paths
- [ ] Covered by existing tests
- [ ] Not required

Performance
- [ ] Tests ran and results are added in the PR description
- [ ] Issue filed with a link in the PR description
- [x] Not required

### CODE REVIEW

File: integration_tests/src/main/python/protobuf_data_gen_test.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        source = spark.createDataFrame(rows, schema)
+        decoded = call_protobuf_function(
+            from_protobuf, f.col("bin"), "test.Simple", desc_path, _simple_desc_bytes)
+        return source.select("i32", "s", decoded.alias("decoded")).collect()
+
+    decoded_rows = with_cpu_session(decode)
+    assert len(decoded_rows) == len(rows)
+    for row in decoded_rows:
+        assert tuple(row.decoded) == (row.i32, row.s)
```
[the remark this answers is not in the document]
Comment (thirtiseven) [reply integration_tests/src/main/python/protobuf_data_gen_test.py:116]: These tests validate a CPU-side test-data helper, which deliberately runs with the plugin disabled. They check schema preservation and that each encoded payload decodes to its corresponding logical row. GPU execution or fallback is not part of this helper’s contract; those assertions belong in the downstream `from_protobuf` operator tests. The existing `data_gen_test.py` similarly tests generator behavior without GPU assertions. The GPU-verification rule should distinguish operator tests from tests of CPU-only test infrastructure.

**Review by firestarman — COMMENTED** [review]
LGTM, but better to have more eyes.

**Review by res-life — APPROVED** [review]
LGTM

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build
